### PR TITLE
 Improve the 'file has changed' dialog

### PR DESCRIPTION
**Pull request by kaiyue0329**
Issue: #5831

#### What it does
- Improve the "file has been changed" dialog

#### How to test
- Need help with steps to reproduce.

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/filesystem/src/browser/filesystem-frontend-module.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default new ContainerModule(bind => {
     bind(FileSystemWatcher).toSelf().inSingletonScope();
     bind(FileShouldOverwrite).toFunction(async function (file: FileStat, stat: FileStat): Promise<boolean> {
         const dialog = new ConfirmDialog({
-            title: `The file '${file.uri}' has been changed on the file system.`,
+            title: `The file '${file.uri.substring(file.uri.lastIndexOf('/') + 1)}' has been changed on the file system.`,
```
Comment (vince-fugnitto) [inline packages/filesystem/src/browser/filesystem-frontend-module.ts]: Thanks! Based on our [coding guidelines](https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#use-short-name), one should use the `labelProvider.getName()` in order to get a proper representation of a filename across each operating system.

**Comment by lmcbout** [comment]
The path would remove ambiguity. Can you add or display it as a tooltip over the dialog ?

**Comment by vince-fugnitto** [comment]
> The path would remove ambiguity. Can you add or display it as a tooltip over the dialog ?

I think a tooltip would be too complicated and un-intuitive for a pop up dialog.
I propose to add the `getLongName()` directly in the body of the dialog as I previously commented.

**Comment by vince-fugnitto** [comment]
@kaiyue0329 do you mind squashing your commits into a single commit, preserving the detailed commit message?

**Comment by lmcbout** [comment]
Looking at the picture, can you make the window smaller by putting the message data in two lines , i think it would be better since the file path is very long

**Comment by vince-fugnitto** [comment]
> Looking at the picture, can you make the window smaller by putting the message data in two lines , i think it would be better since the file path is very long

@lmcbout not sure the change should be applied here. Would it not be better to have a `max-width` for dialogs in general? This means that the application will be responsible for updating the size and not extenders.

Example, I created the following dialog:

<div align='center'>

Screen Shot 2020-01-14 at 2 25 52 PM

</div>

I think this is a separate enhancement that can be tracked in an issue.

**Comment by lmcbout** [comment]
Very very long text could be handle on a separate issue, It is OK for me
Looks good

**Comment by vince-fugnitto** [comment]
@kaiyue0329 I think you might have pushed changes unrelated to this pull request instead of opening a new one.

**Pull request merged**